agent/cache/entry.go

```diff
@@ -26,9 +26,10 @@ type cacheEntry struct {
 	Index uint64
 
 	// Metadata that is used for internal accounting
-	Valid    bool          // True if the Value is set
-	Fetching bool          // True if a fetch is already active
-	Waiter   chan struct{} // Closed when this entry is invalidated
+	Valid       bool          // True if the Value is set
+	Fetching    bool          // True if a fetch RPC is currently active
+	GoroutineID uint64        // Nonzero if a fetch goroutine is running.
```

Comment:
Switched this from a `bool` to a `uint64` so that goroutines can fingerprint their own entry.

This is important in this scenario:

1. bg fetch launches for KEY
2. time passes
3. expiry loop clears entries map for KEY
4. new request re-inserts entry for KEY
5. **new** bg fetch launches for KEY
6. original bg fetch doesn't bail because KEY exists!

Then you end up with two goroutines fetching the same KEY.

This occasionally failed `TestCacheGet_expireBackgroudRefresh`